functions/consumer/rabbit-consumer/src/main/java/org/springframework/cloud/fn/consumer/rabbit/RabbitConsumerConfiguration.java

```diff
@@ -144,25 +146,49 @@ public void destroy() {
-//				CachingConnectionFactory cf = super.rabbitConnectionFactory(config, resourceLoader, credentialsProvider,
-//						credentialsRefreshService, connectionNameStrategy, connectionFactoryCustomizers);
-				cachingConnectionFactory.setConnectionNameStrategy(
-						connectionFactory -> "rabbit.sink.own.connection");
-				cachingConnectionFactory.afterPropertiesSet();
-				return cachingConnectionFactory;
-			}
-
-		}
+	private ConnectionFactory buildLocalConnectionFactory() throws Exception {
+		this.ownConnectionFactory = rabbitConnectionFactory(this.bootProperties, this.resourceLoader,
+				this.credentialsProvider, this.credentialsRefreshService, this.connectionFactoryCustomizers);
+		return this.ownConnectionFactory;
 	}
 
-}
-
-
+	/**
+	 *
+	 * @param properties
+	 * @param resourceLoader
+	 * @param credentialsProvider
+	 * @param credentialsRefreshService
+	 * @param connectionFactoryCustomizers
+	 * @return
+	 * @throws Exception
+	 */
+	private CachingConnectionFactory rabbitConnectionFactory(RabbitProperties properties, ResourceLoader resourceLoader,
+			ObjectProvider<CredentialsProvider> credentialsProvider,
+			ObjectProvider<CredentialsRefreshService> credentialsRefreshService,
+			ObjectProvider<ConnectionFactoryCustomizer> connectionFactoryCustomizers) throws Exception {
+
+		/* NOTE: This is based on RabbitAutoConfiguration.RabbitConnectionFactoryCreator
+		 * 		https://github.com/spring-projects/spring-boot/blob/c820ad01a108d419d8548265b8a34ed7c5591f7c/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/amqp/RabbitAutoConfiguration.java#L95
+		 * [UPGRADE_CONSIDERATION] this should stay somewhat in sync w/ the functionality provided by its original source.
+		 */
+		RabbitConnectionFactoryBean connectionFactoryBean = new RabbitConnectionFactoryBean();
+		RabbitConnectionFactoryBeanConfigurer connectionFactoryBeanConfigurer = new RabbitConnectionFactoryBeanConfigurer(resourceLoader, properties);
+		connectionFactoryBeanConfigurer.setCredentialsProvider(credentialsProvider.getIfUnique());
+		connectionFactoryBeanConfigurer.setCredentialsRefreshService(credentialsRefreshService.getIfUnique());
+		connectionFactoryBeanConfigurer.configure(connectionFactoryBean);
+		connectionFactoryBean.afterPropertiesSet();
+
+		com.rabbitmq.client.ConnectionFactory connectionFactory = connectionFactoryBean.getObject();
+		connectionFactoryCustomizers.orderedStream()
+				.forEach((customizer) -> customizer.customize(connectionFactory));
+
+		CachingConnectionFactory cachingConnectionFactory = new CachingConnectionFactory(connectionFactory);
+		CachingConnectionFactoryConfigurer cachingConnectionFactoryConfigurer = new CachingConnectionFactoryConfigurer(properties);
+		cachingConnectionFactoryConfigurer.setConnectionNameStrategy(cf -> "rabbit.sink.own.connection");
+		cachingConnectionFactoryConfigurer.configure(cachingConnectionFactory);
+		cachingConnectionFactory.afterPropertiesSet();
```

Comment:
All of these manual `afterPropertiesSet()` scare me since they are done without an `ApplicationContext` propagation and during other bean definition.
Although it might be OK for us so far since we just need a lite `CF` instance not tied with `AC`.
If you confirm that tests work well with such an approach, then we won't spend too much time on this and just move on.

Reply:
This is taken from what SB [RabbitAutoConfiguration](https://github.com/spring-projects/spring-boot/blob/main/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/amqp/RabbitAutoConfiguration.java#L127) does. I think its fine because the impl of CF has what it needs to create the instance. But I hear ya, I typically raise eyebrow when I see this too.

Reply:
Well, a bit different though: that `RabbitConnectionFactoryBean` is fully OK. I worry more about this `CachingConnectionFactory` which comes with contract like this:
```
implements ConnectionFactory, DisposableBean, BeanNameAware,
		ApplicationContextAware, ApplicationEventPublisherAware, ApplicationListener<ContextClosedEvent>
```

But again: if we are on the same page and there is no harm that we don't tie it with the `AC`, then it is OK to go without those hooks and callback.